### DISCUSSION TITLE
Missing package 'ansible'

### DISCUSSION THREAD

**Discussion by kmcgrego1307** [vmware/packer-examples-for-vsphere#171] [Help] [answered]
I dug around in the logs of the partially-installed VM that Packer was trying to build, and eventually found this:
Mar  9 14:34:27 ubuntu-server subiquity_event.2211[2211]:     subiquity/Install/install/postinstall/install_ansible: installing ansible
Mar  9 14:34:27 ubuntu-server subiquity_log.2211[18082]: **E: Unable to locate package ansible**
Mar  9 14:34:28 ubuntu-server subiquity_event.2211[2211]:     subiquity/Install/install/postinstall/install_ansible: Command '['systemd-cat', '--level-prefix=false', '--identifier=subiquity_log.2211', '/snap/subiquity/2651/usr/bin/python3', '-m', 'curtin', 'system-install', '-t', '/target', '--', 'ansible']' returned non-zero exit status 100.
Mar  9 14:34:28 ubuntu-server subiquity_event.2211[2211]:   subiquity/Install/install/postinstall: Command '['systemd-cat', '--level-prefix=false', '--identifier=subiquity_log.2211', '/snap/subiquity/2651/usr/bin/python3', '-m', 'curtin', 'system-install', '-t', '/target', '--', 'ansible']' returned non-zero exit status 100.
Mar  9 14:34:28 ubuntu-server subiquity_event.2211[2211]: subiquity/Install/install: Command '['systemd-cat', '--level-prefix=false', '--identifier=subiquity_log.2211', '/snap/subiquity/2651/usr/bin/python3', '-m', 'curtin', 'system-install', '-t', '/target', '--', 'ansible']' returned non-zero exit status 100.

And then it crashes and halts.

I've set up a proxy, but I'm still using the apt sources specified in this project's user-data.pkrtpl.hcl. Why would it not be able to find the ansible package? The other packages (openssh-server, open-vm-tools, cloud-init) all install fine.

**Comment by kmcgrego1307** [comment, marked as answer]
This is interesting - I removed the 'ansible' item and got a lot further. Packer moved on to "Executing Ansible" and from the output it seemed to be running so I guess there was some version of Ansible installed anyway. At
TASK [base : Updating the guest operating system.]
I got
fatal: [default]: FAILED! => {"changed": false, "msg": "Failed to update apt cache: "}

After that failure Packer powered off and removed the VM, so I can't explore it for logs unless there is some way to tell Packer to leave the VM?

**Reply by tenthirtyam** [reply]
The local ansible package is not technically required and can be removed from the packages list. It was originally placed there for `ansible-local` provisioner, but I opted to run all playbooks from the Packer host with the `ansible` provisioner. 

The local packages will be removed in the future update but it's not caused an issue in build tests.

**Reply by tenthirtyam** [reply]
The originating VM is destroyed based on the common variable:

```hcl
common_content_library_destroy     = true
```

**Reply by tenthirtyam** [reply]
Addressed in PR https://github.com/vmware-samples/packer-examples-for-vsphere/pull/172.
